### PR TITLE
Implementation of Bot Base and File Receiver for Enhanced File Handling

### PR DESCRIPTION
**Pull request by funkvay-star**
**Key features include:**

- **Bot Base Setup:** Got the basic framework of our bot up and running.
- **File Receiver:** It can now tell apart different types of messages like links, docs, and videos.
- **Metadata Creation:** The bot also makes detailed info for each file, like who sent it and other file details.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Byte-compiled / optimized / DLL files
+*.pyc
+__pycache__/
+*.py[cod]
+
+# db
+*.sqlite3
```
Comment (Artur-Abalov) [inline .gitignore]: to-remove

File: src/bot/main.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import sys
+from dotenv import dotenv_values
+
+# aiogram modules
+from aiogram import Bot, Dispatcher, types
+
+# our modules
+from file_receiver import FileReceiver
+
+config = dotenv_values('.env')
```
Comment (Artur-Abalov) [inline src/bot/main.py]: move global variables like TOKEN out to separate file like definition py

File: src/bot/file_receiver.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    DOCUMENT_SIZE_LIMIT = 100 * MB  # 100 MB
+    VIDEO_SIZE_LIMIT = 500 * MB  # 500 MB
+    URL_PATTERN = (
+        r'http[s]?://'  # Scheme
+        r'(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]'
+        r'|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+    )
+
+    @staticmethod
+    def is_url(message_content):
```
Comment (Artur-Abalov) [inline src/bot/file_receiver.py]: Potentially method for StringHelper util class

File: src/bot/file_receiver.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            "sender_username": sender_username,
+            "sender_name": sender_name,
+            "message_type": message_type,
+            "timestamp": timestamp,
+            "file_size in bytes": file_size,
+        }
+
+        return metadata
+
+    def print_metadata(self, metadata):
```
Comment (Artur-Abalov) [inline src/bot/file_receiver.py:89]: testing method?

File: src/bot/file_receiver.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        for key, value in metadata.items():
+            print(f"{key}: {value}")
+
+    def determine_message_type(self, message: types.Message):
```
Comment (Artur-Abalov) [inline src/bot/file_receiver.py]: Checking message type in fileReceiver class violates SRP

File: src/bot/file_receiver.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        message_content = message.text or ""
+        if self.is_url(message_content):
+            return "link"
+        elif isinstance(message.document, types.Document):
+            return "document"
+        elif isinstance(message.video, types.Video):
+            return "video"
+        elif message.text:
+            return "text"
```
Comment (Artur-Abalov) [inline src/bot/file_receiver.py]: Not needed

File: src/bot/file_receiver.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    def get_file_size(self, message: types.Message):
+        message_type = self.determine_message_type(message)
+
+        if message_type == "document":
+            return message.document.file_size
+        elif message_type == "video":
+            return message.video.file_size
+        else:
+            # No file size for other types (like link or text)
+            return None
```
Comment (Artur-Abalov) [inline src/bot/file_receiver.py]: Potential NPE

File: src/bot/file_receiver.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        message_type = self.determine_message_type(message)
+        timestamp = message.date.isoformat()
+        file_size = self.get_file_size(message)
+
+        metadata = {
+            "sender_id": sender_id,
+            "sender_username": sender_username,
+            "sender_name": sender_name,
+            "message_type": message_type,
+            "timestamp": timestamp,
```
Comment (Artur-Abalov) [inline src/bot/file_receiver.py]: change variable name, its not actually a timestamp, its ISO formatted date

**Review by Artur-Abalov — DISMISSED** [review]

File: src/bot/file_receiver.py (hunk 4 above)
Comment (funkvay-star) [reply src/bot/file_receiver.py:89]: The method is for testing and useful during this review phase. I plan to remove it before the final merge, after resolving all other discussions.

**Review by Artur-Abalov — APPROVED** [review]
everything fixed

**Pull request merged**